app/src/main/java/org/wikipedia/search/SearchResultsFragment.java

```diff
@@ -438,21 +471,77 @@ public void onAddToList(@NonNull PageTitle title, @NonNull InvokeSource source)
         }
     }
 
-    private final class SearchResultAdapter extends RecyclerView.Adapter<SearchResultItemViewHolder> {
+    private final class SearchResultAdapter extends RecyclerView.Adapter<DefaultViewHolder<View>> {
+
+        private static final int VIEW_TYPE_ITEM = 0;
+        private static final int VIEW_TYPE_NO_RESULTS = 1;
+
+        @Override
+        public int getItemViewType(int position) {
+            return totalResults.isEmpty() ? VIEW_TYPE_NO_RESULTS : VIEW_TYPE_ITEM;
+        }
+
         @Override
         public int getItemCount() {
-            return totalResults.size();
+            return totalResults.isEmpty() ? resultsCountList.size() : totalResults.size();
         }
 
         @Override
-        public SearchResultItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
-            return new SearchResultItemViewHolder(LayoutInflater.from(getContext())
-                    .inflate(R.layout.item_search_result, parent, false));
+        public DefaultViewHolder<View> onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+            if (viewType == VIEW_TYPE_ITEM) {
+                return new SearchResultItemViewHolder(LayoutInflater.from(getContext())
+                        .inflate(R.layout.item_search_result, parent, false));
+            } else {
+                return new NoSearchResultItemViewHolder(LayoutInflater.from(getContext())
+                        .inflate(R.layout.item_search_no_results, parent, false));
+            }
         }
 
         @Override
-        public void onBindViewHolder(@NonNull SearchResultItemViewHolder holder, int pos) {
-            holder.bindItem(pos);
+        public void onBindViewHolder(@NonNull DefaultViewHolder<View> holder, int pos) {
+            if (holder instanceof SearchResultItemViewHolder) {
+                ((SearchResultItemViewHolder) holder).bindItem(pos);
+            } else if (holder instanceof NoSearchResultItemViewHolder) {
+                ((NoSearchResultItemViewHolder) holder).bindItem(pos);
+            }
+        }
+    }
+
+    private class NoSearchResultItemViewHolder extends DefaultViewHolder<View> {
+        NoSearchResultItemViewHolder(@NonNull View itemView) {
+            super(itemView);
+        }
+
+        private ColorStateList accentColorStateList = new ColorStateList(
```

Comment:
There's a much simpler convenience method: `ColorStateList.valueOf(...)`

Reply:
Thanks! This is much better.